lib/hmmlearn/vhmm.py

```diff
@@ -0,0 +1,724 @@
+                else:
+                    self.dof_prior_ = self.dof_prior
+
+            elif self.covariance_type == "tied":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            elif self.covariance_type == "diag":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            elif self.covariance_type == "spherical":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            else:
+                raise ValueError(
+                    f"Unknown covariance_type: {self.covariance_type}")
+            self.dof_posterior_ = np.copy(cluster_counts)
+
+        if self._needs_init("w", "W_k_"):
+            if self.covariance_type == "full":
+                if self.scale_prior is None:
+                    self.W_0_inv_ = np.broadcast_to(
+                        np.identity(self.n_features_) * 1e-3,
+                        (self.n_components, self.n_features_, self.n_features_)
+                    )
+                else:
+                    self.W_0_inv_ = self.scale_prior
+
+            elif self.covariance_type == "tied":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            elif self.covariance_type == "diag":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            elif self.covariance_type == "spherical":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            else:
+                raise ValueError(
+                    f"Unknown covariance_type: {self.covariance_type}")
+
+            # Covariance posterior comes from the estimate of the data
+            cv = np.cov(X.T) + 1E-3 * np.eye(X.shape[1])
+            self.covars_posterior_ = \
+                np.copy(
+                    _utils.distribute_covar_matrix_to_match_covariance_type(
+                    cv, self.covariance_type, self.n_components
+                    )
+                )
+
+            self.W_k_inv_ = self.covars_posterior_ * \
+                    np.asarray(self.dof_posterior_)[:, None, None]
+
+            self.W_k_ = np.linalg.inv(self.W_k_inv_)
+            self.W_0_ = np.linalg.inv(self.W_0_inv_)
+
+    def _get_n_fit_scalars_per_param(self):
+        nc = self.n_components
+        nf = self.n_features
+        return {
+            "s": nc - 1,
+            "t": nc * (nc - 1),
+            "m": nc * nf,
+            "c": {
+                "spherical": nc,
+                "diag": nc * nf,
+                "full": nc * nf * (nf + 1) // 2,
+                "tied": nf * (nf + 1) // 2,
+            }[self.covariance_type],
```

Comment:
Might be nice if this wasn't repeated from `hmm.py`

Reply:
Yes - I plan to address this, and the similar method for the categorical HMM, once the Multinomial/Categorical PR is merged.

Reply:
This is different enough that I think it warrants  keeping it separate.  What do you think?